python/ray/dataframe/dataframe.py

```diff
@@ -2418,9 +2444,31 @@ def rdiv(self, other, axis='columns', level=None, fill_value=None):
     def reindex(self, labels=None, index=None, columns=None, axis=None,
                 method=None, copy=True, level=None, fill_value=np.nan,
                 limit=None, tolerance=None):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        if not columns:
+            return DataFrame()
+        col_idx = [self.columns.get_loc(columns[i])
+                   for i in range(len(columns))
+                   if columns[i] in self.columns]
+
+        if not copy:
+            raise NotImplementedError(
+                "To contribute to Pandas on Ray, please visit "
+                "github.com/ray-project/ray.")
+
+        axis = pd.DataFrame()._get_axis_number(axis) if (axis) else 0
+        if axis == 1 or columns:
+            def row_helper(df, col_idx):
```

Comment:
Since the partitions currently have `RangeIndex` on both rows and columns, this function won't work. Refer to `utils._reindex_helper` and ask @devin-petersohn for details.

Reply:
So I tried to get around that by translating the columns passed in to the helper by using the loc index of the columns desired (col_idx)

Reply:
Removed this implementation of reindex for a future PR